src/twisted/test/test_defer.py

```diff
@@ -1681,6 +1683,37 @@ def test_fromCoroutineRequiresCoroutine(self) -> None:
         for thing in thingsThatAreNotCoroutines:
             self.assertRaises(defer.NotACoroutineError, Deferred.fromCoroutine, thing)
 
+    def test_inlineCallbacksCancelCaptured(self) -> None:
+        """
+        Cancelling an `inlineCallbacks` correctly handles the function catching
+        the `CancelledError`.
+        """
+        cancellerMock1 = Mock()
+        cancellerMock2 = Mock()
+        d1: Deferred[Any] = Deferred(cancellerMock1)
+        d2: Deferred[Any] = Deferred(cancellerMock2)
```

Comment:
I am not a fan of Mock, expecilly one without a specification 

Maybe something like this

```suggestion
        canceller1Calls = []
        canceller2Calls = []
        d1: Deferred[Any] = Deferred(lambda d: canceller1Calls.append(d))
        d2: Deferred[Any] = Deferred(lambda d: canceller2Calls.append(d))
```


My biggest issue with Mock assertion is that for example if you call

```diff
-cancellerMock1.assert_called_once()
+cancellerMock1.asert_called_once()
```

it will still pass